torch/distributed/distributed_c10d.py

```diff
@@ -1494,16 +1490,16 @@ def all_gather_multigpu(
 
 def _object_to_tensor(obj):
     f = io.BytesIO()
-    _pickler(f).dump(obj)
+    torch.save(obj, f)
     byte_storage = torch.ByteStorage.from_buffer(f.getvalue())  # type: ignore[attr-defined]
     byte_tensor = torch.tensor(byte_storage, dtype=torch.uint8)
     local_size = torch.tensor([byte_tensor.numel()], dtype=torch.long)
     return byte_tensor, local_size
 
 
-def _tensor_to_object(tensor, tensor_size):
+def _tensor_to_object(tensor, tensor_size, device=None):
     buf = tensor.numpy().tobytes()[:tensor_size]
-    return _unpickler(io.BytesIO(buf)).load()
+    return torch.load(io.BytesIO(buf), map_location=device)
```

Comment:
Does this create any BC issue? i.e. if the map location is `None`, will the default behavior still apply?

Reply:
If the map location is `None`, then the default behavior applies. This is verified by the fact that the existing tests for `broadcast_object_list()` still pass.